### PR TITLE
Remove HMPO app 

### PR DESCRIPTION
**Pull request by LaiTang-GDS**
## Proposed changes

Remove HMPO app from core front

### What changed

Replaced HMPO app with standard middleware implementations to reduce complexity and have full control on ordering middleware.

Converted yml files to use json for managing language content 

### Why did it change

HMPO app features were not being used and in the recent change to use pino-logger workarounds were needed to co-exist with the HMPO app framework. 
Replacing HMPO app now will ensure we have full control of core-front going forward. 

### Issue tracking
- [PYIC-2105](https://govukverify.atlassian.net/browse/PYIC-2105)

### CODE REVIEW

File: src/views/shared/base.njk [hunk 1]
```diff
@@ -0,0 +1,127 @@
+{% extends "govuk/template.njk" %}
+{% from "govuk/components/cookie-banner/macro.njk" import govukCookieBanner %}
+{% from "govuk/components/phase-banner/macro.njk" import govukPhaseBanner %}
+{% from "govuk/components/back-link/macro.njk" import govukBackLink %}
+
+{% block head %}
+
+    <!--[if !IE 8]><!-->
+    <link href="/public/stylesheets/application.css" rel="stylesheet">
```
Comment (AmritSidhu) [inline src/views/shared/base.njk:9]: should this not be application.scss

File: src/views/shared/base.njk [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        contentLicence: {
+            text: 'general.footer.contentLicence.linkText' | translate | safe
+        },
+        copyright: {
+            text: 'general.footer.copyright.linkText' | translate
+        }
+    }) }}
+{% endblock %}
+
+ - href: "https://signin.account.gov.uk/accessibility-statement"
```
Comment (AmritSidhu) [inline src/views/shared/base.njk]: somehow this came back and can be removed

File: src/views/shared/banner.njk [hunk 3]
```diff
@@ -1,93 +1,80 @@
...
-                href: "#",
-                type: "button",
-                classes:"cookie-hide-button",
-                attributes: {
-                "aria-label": 'govuk.cookie.cookieBanner.cookieBannerHideLink' | translate
-            }
-            }
-        ],
-            hidden: true
+          text: "View cookies",
```
Comment (AmritSidhu) [inline src/views/shared/banner.njk]: needs translating

File: src/views/shared/banner.njk [hunk 4]
```diff
@@ -1,93 +1,80 @@
...
+<p class="govuk-body">{{ 'general.cookie.cookieBanner.paragraph1' | translate }}</p>
+<p class="govuk-body">{{ 'general.cookie.cookieBanner.paragraph2' | translate }}</p>
 {% endset %}
 
 {% set acceptHtml %}
-    <p class="govuk-body">{{ 'govuk.cookie.cookieBanner.cookieBannerAccept.paragraph1' | translate }}<a
-                class="govuk-link"
-                href="{{ 'cookieLocation' | trim }}">{{ 'govuk.cookie.cookieBanner.changeCookiePreferencesLink' | translate }}</a> {{ 'govuk.cookie.cookieBanner.cookieBannerAccept.paragraph2' | translate }}
-    </p>
+  <p class="govuk-body">{{ 'general.cookie.cookieBanner.cookieBannerAccept.paragraph1' | translate }}<a class="govuk-link" href="https://www.gov.uk/help/cookies">{{ 'general.cookie.cookieBanner.changeCookiePreferencesLink' | translate }}</a> {{ 'general.cookie.cookieBanner.cookieBannerAccept.paragraph2' | translate }}</p>
```
Comment (AmritSidhu) [inline src/views/shared/banner.njk]: these links should be https://signin.account.gov.uk/cookies

File: src/views/shared/banner.njk [hunk 5]
```diff
@@ -1,93 +1,80 @@
...
 {% endset %}
 
 {% set rejectedHtml %}
-    <p class="govuk-body">{{ 'govuk.cookie.cookieBanner.cookieBannerReject.paragraph1' | translate }}<a
-                class="govuk-link"
-                href="{{ 'cookieLocation' | trim }}">{{ 'govuk.cookie.cookieBanner.changeCookiePreferencesLink' | translate }}</a> {{ 'govuk.cookie.cookieBanner.cookieBannerReject.paragraph2' | translate }}
-    </p>
+  <p class="govuk-body">{{ 'general.cookie.cookieBanner.cookieBannerReject.paragraph1' | translate }}<a class="govuk-link" href="https://www.gov.uk/help/cookies">{{ 'general.cookie.cookieBanner.changeCookiePreferencesLink' | translate }}</a> {{ 'general.cookie.cookieBanner.cookieBannerReject.paragraph2' | translate }}</p>
```
Comment (AmritSidhu) [inline src/views/shared/banner.njk]: these links should be https://signin.account.gov.uk/cookies

File: src/app.js [hunk 6]
```diff
@@ -23,37 +37,72 @@ if (process.env.NODE_ENV !== "local") {
...
+  req.log = logger.child({
+    requestId: req.id,
+    ipvSessionId: req.session?.ipvSessionId,
+    sessionId: req.session?.id,
+  });
+  next();
 });
 
+app.use(cookieParser());
+//app.use(csurf({ cookie: getCSRFCookieOptions(true) }));
```
Comment (AmritSidhu) [inline src/app.js]: should this be removed

File: src/app.js [hunk 7]
```diff
@@ -63,8 +112,29 @@ router.use((req, res, next) => {
   next();
 });
 
+router.get("/", (req, res) => {
```
Comment (AmritSidhu) [inline src/app.js]: this can be removed

File: src/app.js [hunk 8]
```diff
@@ -63,8 +112,29 @@ router.use((req, res, next) => {
...
 router.use("/ipv", require("./app/ipv/router"));
+
+router.get("/healthcheck", (req, res) => {
+  return res.status(200).send("OK");
+});
+
+app.use(router);
+
+app
+  .listen(PORT, () => {
```
Comment (AmritSidhu) [inline src/app.js:127]: is this still necessary

File: src/config/cookie.js [hunk 9]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+  getCSRFCookieOptions(isProdEnv) {
```
Comment (AmritSidhu) [inline src/config/cookie.js]: is this file still needed if we are removing it from app.js

File: src/views/errors/page-not-found.njk [hunk 10]
```diff
@@ -0,0 +1,19 @@
+{% extends "shared/base.njk" %}
+{% from "govuk/components/button/macro.njk" import govukButton %}
+{% set pageTitleName = 'error.pageNotFound.title' | translate %}
```
Comment (AmritSidhu) [inline src/views/errors/page-not-found.njk:3]: what purpose does setting this pageTitleName when we are setting a header below

Comment (AmritSidhu) [reply src/views/errors/page-not-found.njk:3]: This applys to all views

File: src/views/ipv/pyi-kbv-thin-file.njk [hunk 11]
```diff
@@ -0,0 +1,16 @@
...
+{% block content %}
+  <h1 class="govuk-heading-l">{{'pages.pyiKbvThinFile.header' | translate }}</h1>
+  <p class="govuk-body">{{'pages.pyiKbvThinFile.content.paragraph1' | translate }}</p>
+  <p class="govuk-body">{{'pages.pyiKbvThinFile.content.paragraph2' | translate }}</p>
+  <h2 class="govuk-heading-m">{{'pages.pyiKbvThinFile.content.subHeading' | translate }}</h2>
+  <p class="govuk-body">{{'pages.pyiKbvThinFile.content.paragraph3' | translate }}</p>
+
+  {% include 'shared/journey-next-form.njk' %}
+
+  <p class="govuk-body"><a target="_blank" rel="noopener noreferrer" href="{{'pages.pyiKbvThinFile.content.contactLinkHref' | translate }}" class="govuk-link">{{'pages.pyiKbvThinFile.content.contactLinkText' | translate }}</a></p>
```
Comment (AmritSidhu) [inline src/views/ipv/pyi-kbv-thin-file.njk]: we can make the contact link universal under a single translation field, i believe its a repeated translation for each page

File: src/views/ipv/pyi-technical-unrecoverable.njk [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+{% block content %}
+  <h1 class="govuk-heading-l">{{'pages.pyiTechnicalUnrecoverable.header' | translate }}</h1>
+  <p class="govuk-body">{{'pages.pyiTechnicalUnrecoverable.content.paragraph1' | translate }}</p>
+  <h2 class="govuk-heading-m">{{'pages.pyiTechnicalUnrecoverable.content.subHeading' | translate }}</h2>
+  <p class="govuk-body">{{'pages.pyiTechnicalUnrecoverable.content.paragraph2' | translate }}</p>
+
+  {{ govukButton({
+    "text": button_text|default('pages.pyiTechnicalUnrecoverable.content.govUKHomepageButtonText' | translate, true),
+    "type": "Submit",
+    "href": 'pages.pyiTechnicalUnrecoverable.content.govUKHomepageButtonHref' | translate
```
Comment (AmritSidhu) [inline src/views/ipv/pyi-technical-unrecoverable.njk]: this again can be put in some shared translation as its used multiple times

**Comment by AmritSidhu** [comment]
We need to look where we are manually calling pages. I believe the debug is failing as it needs the file extention

File: src/views/shared/base.njk (hunk 1 above)
Comment (wilsond-gds) [reply src/views/shared/base.njk:9]: The scss files from govuk-frontend and our own assets/scss/application.scss are pre-processed into a single minified css file - browsers can't parse scss currently

File: src/app.js (hunk 6 above)
Comment (LaiTang-GDS) [reply src/app.js]: yes safe to remove

File: src/views/errors/page-not-found.njk (hunk 10 above)
Comment (wilsond-gds) [reply src/views/errors/page-not-found.njk:3]: Currently the page title is output from nunjucks with a load of white space around the content - we can get rid of this by doing something like 
```
{% block pageTitle -%}
    {% if translate("pages."+hmpoPageKey+".serviceNameRequired") == 'true' %}
        {{ translate(hmpoPageTitle) }} – Prove your identity – GOV.UK
    {% else %}
        {{ translate(hmpoPageTitle) }} – GOV.UK
    {% endif %}
{%- endblock %}
```
it's the `{%-` and `-%}` that's doing the work trimming whitespace. It's also available for outputting variables, e.g. `{{- annoyingVariable -}}` [source](https://mozilla.github.io/nunjucks/templating.html#whitespace-control)

Comment (wilsond-gds) [reply src/views/errors/page-not-found.njk:3]: Also yes, there is some stuff that I put in about not showing the service name in the page `<title>` which can probably go

**Review by AmritSidhu — DISMISSED** [review]

**Review by AmritSidhu — APPROVED** [review]
